### PR TITLE
PE-451: getter for driveKey

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/commands/list_folder.ts [hunk 1]
```diff
@@ -46,7 +46,9 @@ new CLICommand({
 				CLI_APP_NAME,
 				CLI_APP_VERSION
 			);
-			children = await arDrive.listPrivateFolder(folderId, password);
+			const driveId = await arDrive.getDriveIdForFolderId(folderId);
```
Comment (matibat) [inline src/commands/list_folder.ts:49]: This fires one extra query to get the `driveID` for the specified folder

Comment (fedellen) [reply src/commands/list_folder.ts:49]: Is there a way for the user to provide the driveKey rather than their password?

Comment (matibat) [reply src/commands/list_folder.ts:49]: There is. I'll mention you at that line

File: src/CLICommand/common_context.ts [hunk 2]
```diff
@@ -50,6 +51,24 @@ export class CommonContext {
 		throw new Error('No wallet file neither seed phrase provided!');
 	}
 
+	public async getDriveKey(driveId: DriveID): Promise<Buffer> {
+		const driveKey = this.getParameterValue(DriveKeyParameter);
+		if (driveKey) {
+			return Buffer.from(driveKey);
+		}
```
Comment (matibat) [inline src/CLICommand/common_context.ts:58]: Here @fedellen: if the user provides its drive key, the wallet and password won't be checked

File: src/commands/list_folder.ts (hunk 1 above)
Comment (fedellen) [reply src/commands/list_folder.ts:49]: Thats true there, but we only enter this `if (wallet && password)`. So when I provide a wallet && driveKey, we'd attempt the public version here

Comment (matibat) [reply src/commands/list_folder.ts:48]: Got ya! I didn't realize that, here the change: 60932cf33681a2b7e1f35043ec96c7d6462908fd

**Review by arielmelendez — APPROVED** [review]
Love the cleanup!

**Pull request merged**